src/yunohost/backup.py

```diff
@@ -165,28 +255,39 @@ def _clean_tmp_dir(retcode=0):
                 else:
                     hooks_filtered.add(hook)
 
-        if not hooks or hooks_filtered:
-            logger.info(m18n.n('backup_running_hooks'))
-            ret = hook_callback('backup', hooks_filtered, args=[tmp_dir],
-                                env=env_var)
-            if ret['succeed']:
-                info['hooks'] = ret['succeed']
-
-                # Save relevant restoration hooks
-                tmp_hooks_dir = tmp_dir + '/hooks/restore'
-                filesystem.mkdir(tmp_hooks_dir, 0750, True, uid='admin')
-                for h in ret['succeed'].keys():
-                    try:
-                        i = hook_info('restore', h)
-                    except:
-                        logger.warning(m18n.n('restore_hook_unavailable',
-                                hook=h), exc_info=1)
-                    else:
-                        for f in i['hooks']:
-                            shutil.copy(f['path'], tmp_hooks_dir)
+        logger.info(m18n.n('backup_running_hooks'))
+
+        # Execute hooks
+        env_dict = self._get_env_var()
+        ret = hook_callback('backup', hooks_filtered, args=[self.collect_dir],
+                            env=env_dict, chdir=self.collect_dir)
+
+        if ret['succeed']:
+            self._commit_path_in_csv(env_dict["YNH_BACKUP_CSV"])
+            self.info['hooks'] = ret['succeed']
+
+            # Save relevant restoration hooks
+            tmp_hooks_dir = 'hooks/restore/'
+            filesystem.mkdir(os.path.join(self.collect_dir, tmp_hooks_dir),
+                             0750, True, uid='admin')
+            for h in ret['succeed'].keys():
+                try:
+                    i = hook_info('restore', h)
+                except:
+                    logger.warning(m18n.n('restore_hook_unavailable', hook=h),
+                                   exc_info=1)
+                else:
+                    for f in i['hooks']:
+                        self._mark_for_backup(f['path'], tmp_hooks_dir)
+        else:
+            # FIXME: support hooks failure
+            pass
 
-    # Backup apps
-    if not ignore_apps:
+    def _collect_apps_files(self, apps=[]):
+        """ Prepare backup for each selected apps """
+
+        if apps is None:
+            return
         # Filter applications to backup
         apps_list = set(os.listdir('/etc/yunohost/apps'))
```

Comment:
Don't know if that's in the scope, but ideally this should be changed to something like `app_list(installed=True)`